### ISSUE TITLE
Error: sizeof-array-div error in pwm driver

### ISSUE THREAD

**Issue by mlasch** [NordicSemiconductor/nrfx#105]
With Zephyr 3.2 and SDK 15.1 I get the following new error message
```
/home/xxx/repos/zephyr/zephyr-workspace/modules/hal/nordic/nrfx/hal/nrf_pwm.h:64:54: error: expression does not compute the number of elements in this array; element type is 'nrf_pwm_values_wave_form_t', not 'uint16_t' {aka 'short unsigned int'} [-Werror=sizeof-array-div]
   64 | #define NRF_PWM_VALUES_LENGTH(array)  (sizeof(array) / sizeof(uint16_t))
      |                                                      ^
/home/xxx/repos/zephyr/zephyr-workspace/xxx/application/xxx/src/rocket_trim.c:686:52: note: in expansion of macro 'NRF_PWM_VALUES_LENGTH'
  686 |                                          .length = NRF_PWM_VALUES_LENGTH(m_pwm_soft_start_table),
```
which makes sense because `sizeof(nrf_pwm_values_wave_form_t)` divided by `sizeof(uint16_t)` is not allowed when `-Werror=sizeof-array-div` is active.

My current workaround is using a custom macro like
```
#define MY_NRF_PWM_VALUES_LENGTH(array)  (sizeof(array) / 2)
```
to get the number of 16-bit elements. However, since the affected code is also used in the examples I guess this should be properly fixed upstream.

Thanks in advance

**Comment by nika-nordic** [comment]
Hi @mlasch ,

just wanted to let you know that we will look into this internally

**Comment by mlasch** [comment]
Sure, I created a minimal example based on the `hello_world` Zephyr sample.

```
diff --git a/samples/hello_world/src/main.c b/samples/hello_world/src/main.c
index 9a90c6a6a0..35e7fcf3a2 100644
--- a/samples/hello_world/src/main.c
+++ b/samples/hello_world/src/main.c
@@ -5,8 +5,15 @@
  */
 
 #include <zephyr/kernel.h>
+#include <nrfx_pwm.h>
+
 
 void main(void)
 {
        printk("Hello World! %s\n", CONFIG_BOARD);
+
+       static nrf_pwm_values_wave_form_t foo[1];
+       int length = NRF_PWM_VALUES_LENGTH(foo);
+
+       printk("%d\n", length);
 }
```
I built the example as follows: `west build -p -d build-nrfx -b nrf52840dk_nrf52840 zephyr/samples/hello_world`

Based on Zephyr tag `v3.3.0` with Zephyr SDK 0.16.0.

**Comment by awojasinski-nordicsemi** [comment]
Hi @mlasch,

We fixed the reported issue in the new nrfx 3.1.0 release.
Thank you on behalf of the team working on nrfx for reporting that error.

**Comment by mlasch** [comment]
Great to hear, thanks!
